var/spack/repos/builtin/packages/veloc/package.py

```diff
@@ -26,40 +27,29 @@ class Veloc(CMakePackage):
     version("1.1", sha256="2bbdacf3e0ce4e7c9e360874d8d85b405525bdc7bd992bdb1f1ba49218072160")
     version("1.0", sha256="d594b73d6549a61fce8e67b8984a17cebc3e766fc520ed1636ae3683cdde77cb")
 
-    # TODO: replace this with an explicit list of components of Boost,
-    # for instance depends_on('boost +filesystem')
-    # See https://github.com/spack/spack/pull/22303 for reference
-    depends_on(Boost.with_default_variants)
     depends_on("libpthread-stubs")
     depends_on("mpi")
+    depends_on("boost")
     depends_on("er")
     depends_on("axl@:0.3.0", when="@:1.5")
-    depends_on("openssl")  # Relies on the OpenSSL crypto library for checksums
-    depends_on("pdsh", when="@master")
-    depends_on("cmake@3.9:", type="build")
+    depends_on("axl@0.5.0:", when="@1.6:")
+    depends_on("openssl")
+    depends_on("cmake@3.10:", type="build")
 
-    with when("@1.6:"):
-        depends_on("axl@0.5.0:")
-        depends_on("redset")
-        depends_on("rankstr")
-        depends_on("shuffile")
-        depends_on("kvtree")
-
-    conflicts("%gcc@:4.9.3")
```

Comment:
Should this still be here?/Should this get a variant to require cxxstd=11/17 based on version?

Reply:
It isn't about the simplicity of it, it is about quietly breaking compatibility with other packages in spack by changing the C++ standard interface of a package without any way to inform downstream packages of that change.

The big question I want to answer is does Veloc change the interface in 1.7 to require C++17, and does it have the ability to generate an interface compatible with C++11?

Below, we are also changing the 1.6 standard from 11 -> 17 which is kind of concerning to me. I missed that in the first review.

Reply:
@kwryankrattiger: I think you are confusing ABI compatibility with C++ standard. We require C++17 standard at compile time for VELOC only since 1.6 (both 1.6 and 1.7 need it). We don't care how the downstream libraries were compiled as long as they are ABI compatible (which depends on the compiler version, not the standard).

Reply:
I am not talking about ABI, I am talking about the API. Some projects will change the API they provide based on configuration. I just want to make sure this isn't introducing c++17 requirements quietly to downstream projects. I have been burned by that kind of behavior before and I just wanted to make sure we weren't doing that here.

Reply:
@kwryankrattiger: got it. In that case, I can confirm that we are not changing the API. We offer both a C and C++ API, and the C++ API is C++11 compliant.